sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -4197,7 +4202,9 @@ case class ArrayExcept(left: Expression, right: Expression) extends ArrayBinaryL
             if (left.dataType.asInstanceOf[ArrayType].containsNull) {
               s"""
                  |if ($array2.isNullAt($i)) {
-                 |  $notFoundNullElement = false;
+                 |  if (!$hashSet.containsNull()) {
+                 |    $hashSet.addNull();
```

Comment:
we can remove the `if` and just do `$hashSet.addNull();`

Reply:
Done